### PR TITLE
sage.{arith,crypto,databases,dynamics,lfunctions,quadratic_forms}: Replace imports from sage.*.all for namespace packages

### LINKED ISSUE #34955: sage.{arith,crypto,databases,dynamics,lfunctions,quadratic_forms}: Replace imports from sage.*.all for namespace packages
<div id="comment:0"></div>

Part of meta-ticket #34201

CC:  @alexchandler100

Component: **refactoring**

Author: **Alex Chandler, Matthias Koeppe**

_Issue created by migration from https://trac.sagemath.org/ticket/34955_

### PR DESCRIPTION
**Pull request by mkoeppe**
### 📚 Description
Fixes #34955

### 📝 Checklist

- [ ] I have made sure that the title is self-explanatory and the description concisely explains the PR.
- [x] I have linked an issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### ⌛ Dependencies

Depends on #35119

### CODE REVIEW

**Comment by tobiasdiez** [comment]
There are build errors here

File: src/sage/crypto/stream.py [hunk 1]
```diff
@@ -12,16 +12,16 @@
...
-
-from sage.crypto.util import random_blum_prime
+from sage.arith.misc import GCD as gcd, power_mod
 from sage.monoids.string_monoid import BinaryStrings
-from sage.arith.all import gcd, power_mod
 from sage.rings.finite_rings.finite_field_constructor import FiniteField
 from sage.rings.finite_rings.integer_mod_ring import IntegerModFactory
 from sage.rings.polynomial.polynomial_element import is_Polynomial
 
+from .cryptosystem import SymmetricKeyCryptosystem
```
Comment (tobiasdiez) [inline src/sage/crypto/stream.py]: What's the reason for these changes?

File: src/sage/databases/findstat.py [hunk 2]
```diff
@@ -1499,7 +1499,8 @@ def __call__(self, elt):
             from sage.repl.preparse import preparse
             try:
                 l = {}
-                code = "from sage.all import *\n" + preparse(self.sage_code())
+                environment = 'sage.all'
+                code = f"from {environment} import *\n" + preparse(self.sage_code())
```
Comment (tobiasdiez) [inline src/sage/databases/findstat.py:1503]: Same comment as in the other PR...

**Review by tobiasdiez — CHANGES_REQUESTED** [review]

File: src/sage/crypto/stream.py (hunk 1 above)
Comment (mkoeppe) [reply src/sage/crypto/stream.py]: .all was replaced as described, and I have cleaned up the imports according to best practices (alphabetic, . imports in a separate block)

File: src/sage/databases/findstat.py (hunk 2 above)
Comment (mkoeppe) [reply src/sage/databases/findstat.py:1503]: Likewise.

File: src/sage/crypto/stream.py (hunk 1 above)
Comment (tobiasdiez) [reply src/sage/crypto/stream.py]: Sorry, I meant the change from "from sage.crypto.util import random_blum_prime" to `.util`.

Comment (mkoeppe) [reply src/sage/crypto/stream.py]: It improves consistency because other modules from this package are already imported in this way.

I made such clean-up changes to the imports in several file swhen the .all replacement already created big diffs.

Comment (tobiasdiez) [reply src/sage/crypto/stream.py]: Can we also replace them consistently to absolute imports? There are a few subtle issues with relative imports (e.g. https://stackoverflow.com/questions/16981921/relative-imports-in-python-3) and we do hit them at https://github.com/sagemath/sage/issues/33783.

Comment (mkoeppe) [reply src/sage/crypto/stream.py]: Thanks for reminding me about this problem. Sure, I'll do that.

Comment (tobiasdiez) [reply src/sage/crypto/stream.py]: Thanks!

Comment (mkoeppe) [reply src/sage/crypto/stream.py]: OK, done.

**Comment by mkoeppe** [comment]
There's no work item to be done.

**Comment by tobiasdiez** [comment]
https://github.com/sagemath/sage/pull/35106#discussion_r1105234712

**Comment by mkoeppe** [comment]
Yes, I have already explained to you what these changes are doing and why they are in scope.

File: src/sage/databases/findstat.py (hunk 2 above)
Comment (dimpase) [reply src/sage/databases/findstat.py:1503]: this reads cryptic to me - what's the issue here?

Comment (tobiasdiez) [reply src/sage/databases/findstat.py:1503]: Sorry for the missing details. See https://github.com/sagemath/sage/pull/35105. My issue is that this is a (almost trivial, but well) runtime change to cover up a bug in the regex that is used to detect this all imports.

Comment (dimpase) [reply src/sage/databases/findstat.py:1503]: Let up open a separate issue for this.

**Review by dimpase — APPROVED** [review]
lgtm
